javalib/src/main/scala/scala/scalanative/nio/fs/UnixPath.scala

```diff
@@ -16,37 +16,49 @@ class UnixPath(private val fs: UnixFileSystem, private val rawPath: String)
   override def getFileSystem(): FileSystem =
     fs
 
-  override def isAbsolute(): Boolean =
+  override def isAbsolute(): Boolean = _isAbsolute
+
+  lazy val _isAbsolute =
```

Comment:
Have you considered writing cached implementation manually thus avoiding synchronization costs of `lazy val`s?
Not sure if it would still be correct though or more performant.

Reply:
I've thought about it, but my reasoning for the state of the current patch is:

- lazy val expresses the intent concisely and explicitly.

- the speed of lazy val initialization is implementation dependent. In the JVM implementation of scala, they've been able to largely mitigate the synchronization cost: http://docs.scala-lang.org/sips/pending/improved-lazy-val-initialization.html

- I could be wrong, but it seems that any correct/performant version that I can create for my own cached implementation would be also the same as writing an optimized implementation for lazy val for scala-native (if it is not already optimized).